src/state/routing/slice.ts

```diff
@@ -76,8 +77,9 @@ function getRoutingAPIConfig(args: GetQuoteArgs): RoutingConfig {
     !args.uniswapXEnabled ||
     (args.userDisabledUniswapX && routerPreference !== RouterPreference.X) ||
     (tokenOutIsNative && !uniswapXEthOutputEnabled) ||
-    tradeType === TradeType.EXACT_OUTPUT ||
-    !isUniswapXSupportedChain(tokenInChainId)
+    (!uniswapXExactOutputEnabled && tradeType === TradeType.EXACT_OUTPUT) ||
+    !isUniswapXSupportedChain(tokenInChainId) ||
+    routerPreference === INTERNAL_ROUTER_PREFERENCE_PRICE
```

Comment:
is this `routerPreference === INTERNAL_ROUTER_PREFERENCE_PRICE` line an unrelated change?

Reply:
no its not! price quotes are exact_out enabling exact_out means we need to filter price quotes out specifically